lock-api-objects/src/main/java/com/palantir/lock/watch/TransactionsLockWatchEvents.java

```diff
@@ -25,53 +25,9 @@
  * Represents a condensed view of lock watch events occurring between some known version and a set of start transaction
  * calls.
  */
+@Value.Immutable
 public interface TransactionsLockWatchEvents {
-    <T> T accept(Visitor<T> visitor);
-
-    interface Visitor<T> {
-        T visit(Events success);
-        T visit(ForcedSnapshot failure);
-    }
-
-    static Events success(List<LockWatchEvent> events, Map<Long, Long> startTsToSequence) {
-        return ImmutableEvents.of(events, startTsToSequence);
-    }
-
-    static ForcedSnapshot failure(LockWatchStateUpdate.Snapshot snapshot) {
-        return ImmutableForcedSnapshot.of(snapshot);
-    }
-
-    /**
-     * A successful result contains a list of all lock watch events occurring between the last known version and the
-     * last started transaction, and a mapping of start timestamps to their respective last occurred event.
-     */
-    @Value.Immutable
-    interface Events extends TransactionsLockWatchEvents {
-        @Value.Parameter
-        List<LockWatchEvent> events();
-        @Value.Parameter
-        Map<Long, Long> startTsToSequence();
-
-        @Override
-        default <T> T accept(Visitor<T> visitor) {
-            return visitor.visit(this);
-        }
-    }
-
-    /**
-     * A failure denotes that it was not possible to compute the result for all of the requested transactions. Since
-     * that generally implies we will fail to get the necessary information for the commit timestamp anyway, instead of
-     * giving partial information, we return a single snapshot that should be used to reseed the state of lock watches
-     * for all future transactions.
-     */
-    @Value.Immutable
-    interface ForcedSnapshot extends TransactionsLockWatchEvents {
-        @Value.Parameter
-        LockWatchStateUpdate.Snapshot snapshot();
-
-        @Override
-        default <T> T accept(Visitor<T> visitor) {
-            return visitor.visit(this);
-        }
-    }
+    List<LockWatchEvent> events();
+    Map<Long, IdentifiedVersion> startTsToSequence();
+    boolean clearCache();
```

Comment:
I see why we'd want a boolean here, though this class feels like it should be renamed - it's quite a bit more than just the events now. Maybe `TransactionsLockWatchUpdate`?

Reply:
I think that is reasonable

Reply:
Is that renaming considered a breaking change?

Reply:
Technically yes (though this is internal API so I wouldn't be too concerned).